lib/kafka/cluster.rb

```diff
@@ -46,6 +46,10 @@ def add_target_topics(topics)
         @target_topics.merge(new_topics)
 
         refresh_metadata!
+
+        cluster_info.topics.each do |topic|
+          Protocol.handle_error(topic.topic_error_code)
+        end
```

Comment:
Also, this could potentially break user code in some cases where the Kafka error is transient... I think the current error handling code handles recovery, but this snippet does not.